app/javascript/packs/containers/ReviewModeContainer.jsx

```diff
@@ -7,10 +7,89 @@ import React from 'react';
 
 import QuestionContainer from './QuestionContainer';
 
+import { getAnswerForQuestionAndBuilding } from '../selectors/answersSelector';
+import { getDependentQuestionsForOption } from "../selectors/questionsSelector";
 import { connect } from 'react-redux';
 import { getQuestionsByBuilding } from '../selectors/questionsSelector';
 
+import { post, patch } from '../fetch/requester';
+
+async function postDelegations(delegations, callback) {
+  var body = {
+    "delegations": delegations,
+  };
+
+  try {
+    let resp = await post('/api/delegations', body);
+    callback(true);
+  } catch (error) {
+    callback(false);
+  }
+}
+
 class ReviewModeContainer extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      status_string: "Delegations have not been saved yet.",
+    }
+  }
+
+  // recursively find all dependent questions, return a list
```

Comment:
Can we break this into the `utils` folder, for @arpanpal's use?

Reply:
@arpanpal 

I don't think it is a good idea to turn this into a utility function. It is a class method now.

If this method is to be turned into a utility function, it needs to accept the following list of input:

parentQuestion -- a question object
getDependentQuestion -- a function object that accepts option id and question_type; user must encapsulate a selector with certain building id and pass it in

The interface feels too complex to me as a utility function. The complexity comes from the fact that global state is only accessible by selectors.

If you still want me to do it, let me know.

Reply:
Would it make sense as its own selector then, to be shared/imported by other components?

Reply:
![image](https://user-images.githubusercontent.com/2977329/38445576-f8a2b2fe-39a8-11e8-94ba-cad3d826640c.png)

This won't be a performance issue